test/fixtures/rules/std_rule_cases/L070.yml

```diff
@@ -0,0 +1,482 @@
+        FROM table2
+      )
+      SELECT *
+      from cte2 as cte_al
+    )
+    UNION
+    SELECT e, f
+    FROM table3
+
+test_pass_known_number_of_result_columns_17:
+  pass_str: |
+    SELECT *
+    FROM table1 as table2
+    UNION
+    SELECT e, f
+    FROM table3
+
+test_fail_unknown_number_of_result_columns_7:
+  fail_str: |
+    SELECT *
+    FROM
+    (
+      WITH cte2 AS (
+        SELECT a
+        FROM table2
+      )
+      SELECT *
+      from cte2 as cte_al
+    )
+    UNION
+    SELECT e, f
+    FROM table3
+
+test_pass_nested_ctes_1:
+  # Test for issue 1984
+  pass_str: |
+    with a as (
+        with b as (
+            select 1 from c
+        )
+        select * from b
+    )
+
+    select * from a
+    union all
+    select k
+    from t2
+
+test_fail_nested_ctes_1:
+  # Test for issue 1984
+  fail_str: |
+    with a as (
+        with b as (
+            select a from c
+        )
+
+        select * from b
+    )
+
+    select * from a
+    union
+    select a, b
+    from t2
+
+test_cte_reference_outer_2:
+  pass_str:
+    with a as (
+        select 1 from b
+    )
+
+    select * from (
+        select * from a
+    )
+
+    union
+
+    select 2
+    from c
+
+test_cte_tricky_nesting_3:
+  pass_str:
+    with b as (
+        select 1 from c
+    )
+
+    select * from (
+        with a as (
+            select * from b
+        )
+
+        select * from a
+    )
+
+    union
+    select a
+    from t2
+
+test_nested_and_same_level_ctes_4:
+  pass_str:
+    with a as (
+      with c as (
+        select 1 from d
+      ),
+
+      b as (
+        select * from c
+      )
+
+      select * from b
+    )
+
+    select * from a
+    union
+    select k
+    from t2
+
+test_nested_cte_references_outer_5:
+  pass_str:
+    with c as (
+      select 1 from d
+    ),
+
+    a as (
+      with b as (
+        select * from c
+      )
+
+      select * from b
+    )
+
+    select * from a
+    union
+    select k
+    from t2
+
+test_pass_join_inside_cte_with_unqualified:
+  pass_str:
+    with cte as (
+        select
+            *
+        from
+            t1
+        inner join t2
+    )
+
+    select a, b from cte
+    union
+    select c, d from cte2;
+
+test_pass_known_number_of_columns_in_two_join_subqueries:
+  pass_str:
+    select
+        *
+    from (
+        select a
+        from foo
+    ) t1
+    inner join (
+        select b
+        from bar
+    ) t2
+    union
+    select c, d from t3;
+
+test_fail_two_join_subqueries_one_with_unknown_number_of_columns:
+  fail_str:
+    select
+        *
+    from (
+        select b
+        from foo
+    ) t1
+    inner join (
+        select b
+        from bar
+    ) t2
+    union
+    select c, d, e from t3;
+
+
+test_query_on_snowflake_stage:
+  pass_str:
+    select mycolumn1
+    from @public.mytable1
+    union
+    select mycolumn2
+    from table2
+  configs:
+    core:
+      dialect: snowflake
+
+
+test_pass_cte_no_select_final_statement:
```

Comment:
Should we add a similar failure case (i.e. where the SELECT is inside an UPDATE)?